cpp/bindings/bindings.hpp

```diff
@@ -53,6 +62,73 @@ class PyPiecewisePolyPath {
   std::string __repr__();
 };
 
+/**
+   Wrapper for TOPPRA Algorithm.
+
+   TODO: Make this class a generic wrapper.
+ */
+class PyTOPPRA {
+ public:
+  PyTOPPRA(LinearConstraintPtrs constraints, PyPiecewisePolyPath& path);
+  PyTOPPRA(const py::list& constraints, PyPiecewisePolyPath& path);
+  ReturnCode computePathParametrization(value_type vel_start = 0,
+                                        value_type vel_end = 0);
+  ParametrizationData getParameterizationData() const;
+  void setN(int N);
+
+ private:
+  std::unique_ptr<toppra::algorithm::TOPPRA> m_problem;
+  LinearConstraintPtr m_tmp;
+};
+
+/**
+   Base Python wrapper for constraints.
+ */
+class PyLinearConstraint {
```

Comment:
Why do you need to wrap LinearConstraint and TOPPRA ?
I think you can directly work with LinearConstraint, no ? In case the issue is memory storage, see https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html#std-shared-ptr

Reply:
I tried working directly before this, it doesn't work. It's because the TOPPRA constructor looks for LinearConstraintPtrs. Yet, when I give the python bindings a list of different constraints, it is not cast properly to LinearConstraintPtrs.

Reply:
I think you must declare the LinearConstraint as
```
py::class_<LinearConstraint, LinearConstraintPtr> (...)
```

Reply:
And you have to do the same for all the constraint classes.